evalml/automl/automl_search.py

```diff
@@ -67,7 +67,9 @@
 class AutoMLSearch:
     """Automated Pipeline search."""
     _MAX_NAME_LEN = 40
+    _MAX_TRAINING_ROWS = int(1e5)
     _LARGE_DATA_ROW_THRESHOLD = int(1e5)
+    _LARGE_DATA_PERCENT_TEST = 0.75
```

Comment:
I can see how the name can be confusing; I'll rename it to `_LARGE_DATA_PERCENT_VALIDATION`.

Re: setting it as a constructor arg, I purposefully make it hard to redefine this variable. My reasoning is that if a user wanted to change the percentage, they could just define their own `data_split`. Especially since this is only applicable to large datasets, it might be confusing as what constitutes a "large" dataset (and the value of which is a class property anyways) and thus when it would kick in. 

It might be worthwhile to update the docs to include a section on how we handle large datasets. Here, it might make sense to describe this default behavior and mention defining a new data_splitter as the best way to redefine this percentage.

Reply:
@christopherbunn thanks for following up!

RE the docs, I agree. We're currently missing a [User Guide](https://evalml.alteryx.com/en/stable/user_guide/automl.html) page describing the automl algorithm, and the heuristics we use to make decisions about the automl algorithm like what data splitting strategy to use and when we use CV vs train-validation splitting. We will get around to adding that. However, I don't think that information is critical for users. Users need to be able to pass in data, get models and select one; the details of how the models were discovered is only of interest to advanced users.

RE constructor arg: gotcha, I follow. The advantage of adding it as a "private" constructor arg is that we don't need to preserve backwards compatibility for it if we decide to remove it in the future, but we can easily change it in our perf tests, or direct users on how to change it. Not required to do it that way, but I think it'd keep us flexible.